src/groupeddataframe/splitapplycombine.jl

```diff
@@ -269,6 +268,7 @@ Last Group: 1 row
 See [`by`](@ref) for more examples.
 """
 function Base.map(f::Function, gd::GroupedDataFrame)
+    @nospecialize
```

Comment:
I hadn't realized you really wanted to add this to all functions. Probably better discuss this separately: if we change our policy and mark non performance-critical functions with `@nospecialize`, many more files will have to be touched. And anyway it adds noise to the PR.

For now, probably better only mark with `@nospecialize` arguments which really change type on (almost) each invokation:
```suggestion
function Base.map(@nospecialize(f::Function), gd::GroupedDataFrame)
```
Though since `f` isn't called I think the default will be equivalent. So maybe not add `@nospecialize` at all here and in similar methods, which makes the code quite simpler.

Also, using `@nospecialize` only on arguments that need it allows keeping one-liners.

Reply:
OK - I will minimize `@nospecialize` in this PR.